examples/python/mini_ic3.py

```diff
@@ -97,12 +104,24 @@ def subst_vars(self, prefix, inv, fml):
 
     def mk_subst(self, prefix, inv):
         self.index = 0
-        return [(f, self.mk_bool(prefix)) for f in inv.children()]
+        if self.is_inv(inv) is not None:
+            return [(f, self.mk_bool(prefix)) for f in inv.children()]
+        else:
+            vars = self.get_vars(inv)
+            return [(f, self.mk_bool(prefix)) for f in vars]
 
     def mk_bool(self, prefix):
         self.index += 1
         return Bool("%s%d" % (prefix, self.index))
 
+    def get_vars(self, f, rs=[]):
```

Comment:
possibly use a table to remember which terms were visitied and avoid recursing over the same sub-terms.